docs/docs/guides/build/components/state-backed-components/managing-state-in-production.md

```diff
@@ -0,0 +1,316 @@
+---
+title: 'Managing state in production'
+description: Learn how to refresh and manage component state in production deployments using the dg CLI.
+sidebar_position: 200
+---
+
+import Beta from '@site/docs/partials/_Beta.md';
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Beta />
+
+Managing state in production depends on which state management strategy you've configured for your state-backed components. This guide explains how to refresh state for both Local Filesystem and Versioned State Storage strategies.
+
+## Before you begin
+
+Ensure your project is installed in the current environment (e.g. GitHub Actions, Docker, etc.) where you'll run state refresh commands.
+
+<Tabs groupId="package-manager">
+<TabItem value="uv" label="uv">
+
+```bash
+uv pip install -e .
+```
+
+</TabItem>
+<TabItem value="pip" label="pip">
+
+```bash
+pip install -e .
+```
+
+</TabItem>
+</Tabs>
+
+## Refreshing state for Local Filesystem components
+
+If your components use the `LOCAL_FILESYSTEM` state management type, you typically refresh state during your CI/CD pipeline as part of building your Docker image.
+
+### Basic refresh command
+
+Run this command to refresh state for all Local Filesystem components:
+
+```bash
+dg utils refresh-defs-state --management-type LOCAL_FILESYSTEM
+```
+
+This command:
+1. Discovers and loads all state-backed components in your project
+2. Filters to only those using `LOCAL_FILESYSTEM` management type
+3. Fetches fresh state from each external system (Tableau, Fivetran, etc.)
+4. Writes the state to the `.local_defs_state` directory in your project
+5. Reports success or failure for each component
+
+### Example: Dockerfile with state refresh
+
+Here's a Dockerfile that refreshes state during the image build process:
+
+```dockerfile
+FROM python:3.11-slim
+
+WORKDIR /app
+
+# Copy project files
+COPY . /app/
+
+# Install dependencies
+RUN pip install -e .
+
+# Refresh component state during build
+RUN dg utils refresh-defs-state --management-type LOCAL_FILESYSTEM
+```
+
+This approach:
+1. Installs your project and its dependencies
+2. Runs `dg utils refresh-defs-state` to fetch fresh state from external APIs
+3. Stores the state in the `.local_defs_state` directory within the image
+
+This causes the state to become part of your deployment artifact.
+
+:::info .local_defs_state and version control
+
+The `.local_defs_state` directory is automatically excluded from version control via an auto-generated `.gitignore` file. However, it **should** be included in your Docker image as part of your deployment artifact.
+
+:::
+
+### Refreshing specific components
+
+To refresh only specific component state keys:
```

Comment:
Should we add guidance here on how to determine the values of the auto-generated keys?

Reply:
Reasonable point, but I actually rethought this and I think including this info really just muddies the waters -- the usecase of only refreshing a subset of the keys is pretty niche and users that find themselves in that scenario can just read the help text of the cli command. Don't want to imply that doing this is a normal or recommended workflow so just removed it.